### ISSUE TITLE
ci: improve master builds

### ISSUE THREAD

**Issue by yakimant** [status-im/status-app#11901]
### Description

Upd 17/08/2023: Added option 4

#### Situation

We have a job in Jenkins running build & e2e tests on each push to master:
https://ci.infra.status.im/job/status-desktop/job/master/

We also run build & e2e on each PR branch push.
Build failure blocks PR from merging, but being up-to-date with master is not required.

We also run e2e nightly on `master`:
https://ci.infra.status.im/job/status-desktop/job/nightly-e2e/

#### Usage

Developers are not requiring PR branch to be up-to-date with` master`, because it becomes difficult to merge.
But still they want to know, that `master` is not broken.
Previously they've got notifications on Discord, but it's not the case with switch to Status.
Currently noone is watching the results manually, unless maybe when `master` got broken for bisect.

#### Diff with mobile flow

1. Mobile don't run e2e on each push, only on demand
2. Mobile require PR to be up-to-date with `master`
3. Thus they know for sure, that code doesn't break `master`
4. And don't need `master` builds

#### Options

1. Leave `master` build, enable notifications to Status
2. Switch flow to Mobile-like
3. Switch off `master` build
4. Leave as is, useful for failure bisecting

**Comment by osmaczko** [comment]
I'll opt for Option 1.

For Option 2, as you mentioned, it will make PRs hard to merge.

Regarding Option 3, I believe pushes to master are much less frequent than pushes to certain PRs. Contributors often modify the code to address review suggestions and fix bugs found by QAs. In other words, removing the master build will not significantly improve the load on Jenkins' nodes.

There is also Option 4 - do nothing. Pros: Enables easier bisecting when things go sideways.. Cons: It has some impact on the load on Jenkins' nodes.

**Comment by yakimant** [comment]
1. As we are back to Discord, let's have master/nightly failures alerts to Discord.

2. Looks like developers don't like the idea of being up-to-date with master before merge.
And issues are not often.

3. As for e2e - I would go for running on demand, not automatically.
But if load is not high - let's keep it.

**Comment by jrainville** [comment]
@yakimant is this still needed?

**Comment by yakimant** [comment]
No. We switched from status to discord, run master nightly and have discord notifications.

Maybe e2e runs need optimisation, but thats out of this issue.

Thanks!
